pkg/services/nvd.go

```diff
@@ -51,45 +51,29 @@ const (
 )
 
 func fetchNvdDataByCPE(cpe string, baseNvdAPIURL string) (*dto.NvdAPIResponse, error) {
-	// Use custom http client with a timeout
-	client := createNVDHTTPClient()
-
-	// Build URL
-	query := url.Values{}
-	query.Set("cpeName", cpe)
-	query.Set("isVulnerable", "")
-	apiURL := baseNvdAPIURL + "?" + query.Encode()
-
-	var nvdResponse *dto.NvdAPIResponse
-	var err error
-
-	for attempt := 0; attempt <= maxRetries; attempt++ {
-		nvdResponse, err = attemptFetch(client, apiURL)
-
-		// Success case
-		if err == nil {
-			return nvdResponse, nil
+	var currentResponse *dto.NvdAPIResponse
+	for {
+		nextResponse, err := fetchNextNvdPage(currentResponse, baseNvdAPIURL, cpe)
+		if nextResponse == nil && err == nil {
+			slog.Info("No more pages to fetch")
```

Comment:
Should be slog.Debug 
```suggestion
			slog.Debug("No more pages to fetch")
```